[리액트 7기 표후동] 8주차_더 나아가기

ndopy opened the pull request:
안녕하세요! 8주차도 잘 부탁드립니다.

#### 진행된 항목
- Redux Toolkit 을 이용한 리팩토링

#### 진행할 항목
- PureComponent 적용해보기
- CSS-in-JS 로 레스토랑 앱 꾸미기

hannut91 committed:
Move index.html file to public directory

Commit:
Build: browserslist update

Commit:
Build: install @reduxjs/toolkit

Commit:
Refactor: reducer to slice

Commit:
Refactor: get actions from slice and export action creators

Commit:
Fix: slice 의 reducer 의 payload 수정

- payload 를 직접 받는 형식?으로 작성해준다.

Commit:
Fix: createSlice 생성 시 지정한 name 을 action type 마다 붙여준다.

Commit:
Refactor: actions 의 내용을 slice 로 옮긴다

Commit:
Refactor: actions.js 에서 import 하던 내용들을 slice.js 에서 import 하도록 수정한다.

Commit:
Refactor: actions.js 에서 import 하던 내용들을 slice.js 에서 import 하도록 수정한다.

Commit:
Remove reducer.js

Commit:
Build: uninstall redux-thunk

Commit:
Fix: 로그인 페이지에서 그냥 로그인이 되는 버그 수정

- 로컬 스토리지에 아무것도 없을 때 load 를 하면 'undefined' 가 있는데, 이것을 액세스 토큰으로 인식하고 있다고 한다.

Commit:
Refactor: slice.js

- 사용하지 않는 thunk 및 middleware import 문 제거
- createStore -> configureStore

moonkii commented on src/actions.js:
```diff
@@ -9,47 +9,35 @@ import {
-    payload: { categoryId },
-  };
-}
+import {
+  setRegions,
+  setCategories,
+  setRestaurants,
+  setRestaurant,
+  selectRegion,
+  selectCategory,
+  changeLoginField,
+  setAccessToken,
+  logout,
+  changeReviewField,
+  clearReviewFields,
+  setReviews,
+} from './slice';
+
+export {
+  setRegions,
+  setCategories,
+  setRestaurants,
+  setRestaurant,
+  selectRegion,
+  selectCategory,
+  changeLoginField,
+  setAccessToken,
```
redux Toolkit 을 사용하면 더이상 action 파일은 필요하지 않을거에요.

moonkii commented on src/slice.js:
```diff
@@ -0,0 +1,233 @@
+  };
+}
+
+export default reducer;
```
하나의 slice 가 복잡한 경우 도메인따라 여러 slice 로 나누어서 사용할 수 있어요!
combineReducer 를 찾아보시면 도움되실거에요 ;)

ndopy replied:
`combineReducer` 에 대해 검색해서 공부하고 적용해보겠습니다.

moonkii reviewed (commented):
<img width="536" alt="image" src="https://user-images.githubusercontent.com/46776329/167972556-4c9b5107-3523-4ec7-ab94-3c386ccf5388.png">

하나씩 잘 해나가고 있으신 것 같아요!
빠르게 완성해보는걸 목표로 해보시면 좋을 것 같네요 ;)

Commit:
Refactor: React.memo 적용하기

Commit:
Add: babel.config.js 에 'import React' 문을 생략하는 옵션을 추가한다.

Commit:
Add: useCallback 적용하기

- if 문은 useCallback 밑에 위치시킨다.

Commit:
Refactor: <RestaurantDetail />

- React.memo 적용하기

Commit:
Refactor: <LoginContainer />, <LoginForm />

- useCallback 과 memo 를 적용한다.

Commit:
Build: install @emotion/styled @emotion/react

- Emotion V11 부터 @emotion/core -> @emotion/react 로 rename 됐다고
  한다.
  - 출처 : https://emotion.sh/docs/emotion-11

Commit:
Style: Emotion 을 이용한 CSS 적용하기

Commit:
Style: Emotion 을 이용한 CSS 적용하기

Commit:
Remove: action.js, action.test.js

hdpyo commented:
전체 강의는 끝났지만 아직 많은 부분을 이해 못해서 부족한 점이 많은 걸 느낍니다.

강의와 참고 자료들을 반복해서 보며 내용을 이해하고,
기존 코드/내용을 최대한 보지 않으면서 인출하려고 많이 시도해야겠고
앞으로 진행할 포트폴리오 과정에서는 꼭 적용해려고 합니다.

moonkii commented:
8주동안 배우신 내용들 활용해서 개인적으로 프로젝트를 꼭 해보세요.
반복 숙달해서 숙련도를 높이고 나면 응용까지도 가능해질거에요 ;)
8주동안 고생 많으셨어요 👍
